README.md

```diff
@@ -1,2 +1,299 @@
-# helm-mi
-This Repo will use to maintain MI related helms
+# Kubernetes Deployment Guide for WSO2 Micro Integrator (MI) and Integration Control Plane (ICP)
+
+This guide provides step-by-step instructions to deploy WSO2 Micro Integrator (MI) and Integration Control Plane (ICP) in a Kubernetes environment using Helm charts. Both MI and ICP will be deployed in the same namespace for seamless integration.
+
+---
+
+## Prerequisites
+
+1. **Kubernetes Cluster**: Ensure you have an operational Kubernetes cluster (e.g. AKS, EKS, GKE, or a local Kubernetes cluster).
+2. **Helm**: Install Helm (version 3 or later) on your local machine.
+3. **Ingress Controller**: Deploy an ingress controller (e.g. [NGINX Ingress Controller](https://kubernetes.github.io/ingress-nginx/deploy/)).
+4. **Docker Images**: Acquire WSO2 product Docker images. Push them to a container registry (e.g. ACR, ECR, GCR) if necessary.
+5. **WSO2 Subscription**: A valid WSO2 subscription is required to access Docker images from the WSO2 private registry.
+
+---
+
+## Deploy a Local Setup
+
+### 1. Set Up a Local Kubernetes Cluster
+
+You can utilize either **Minikube** or **Rancher Desktop** to establish a local Kubernetes cluster:
+
+#### Using Minikube
+
+1. Install Minikube by following the [official documentation](https://minikube.sigs.k8s.io/docs/start/).
+2. Start a Minikube cluster:
+   ```bash
+   minikube start --cpus=4 --memory=8192
+   ```
+3. Enable the [NGINX ingress](https://kubernetes.github.io/ingress-nginx/deploy/#minikube) controller:
+   ```bash
+   minikube addons enable ingress
+   ```
+
+#### Using Rancher Desktop
+
+1. Install Rancher Desktop from the [official website](https://rancherdesktop.io/).
+2. Configure Rancher Desktop to use Kubernetes and set the desired Kubernetes version.
+3. Ensure the [NGINX ingress](https://kubernetes.github.io/ingress-nginx/deploy/#rancher-desktop) controller is installed.
+
+### 2. Clone the Helm Chart Repository
+Clone the Helm chart repository containing the MI and ICP Helm charts:
+```bash
+git clone https://github.com/wso2/helm-mi.git
+git fetch --all
+```
+
+Checkout to your preferred MI runtime version branch:
+```bash
+git checkout origin/4.3.x
+cd helm-mi
```

Comment:
Shouldn't we do `cd` before the `git checkout`?

Reply:
modified steps by https://github.com/wso2/helm-mi/pull/18/commits/81752588a0b25ac6187cb8296525d2bd4bc2464a